### PR TITLE
Add a small python script to verify common mapping mistakes

### PR DESCRIPTION
**Pull request by lzimann**
Not the best code in the world, but does the trick I guess. 
It tests for: map not being tgm, variables we do not want(step_x/y, d1/d2), double turfs/areas on the same key.

### CODE REVIEW

File: tools/travis/build_byond.sh [hunk 1]
```diff
@@ -5,29 +5,13 @@ set -e
 shopt -s globstar
 
 if [ "$BUILD_TOOLS" = false ]; then
-	if grep '"aaa" = \(.+\)' _maps/**/*.dmm;	then
```
Comment (ShizCalev) [inline tools/travis/build_byond.sh]: Keep this one in.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: tools/travis/build_byond.sh [hunk 2]
```diff
@@ -5,29 +5,13 @@ set -e
...
 	fi;
-	if grep '\td[1-2] =' _maps/**/*.dmm;	then
-    	echo "d[1-2] cable variables detected in maps, please remove them."
-    	exit 1
-	fi;
 	if grep '^/area/.+[\{]' _maps/**/*.dmm;	then
     	echo "Vareditted /area path use detected in maps, please replace with proper paths."
     	exit 1
 	fi;
-	if grep '\W\/turf\s*[,\){]' _maps/**/*.dmm; then
```
Comment (ShizCalev) [inline tools/travis/build_byond.sh]: Keep this one too.

File: tools/travis/build_byond.sh (hunk 1 above)
Comment (lzimann) [reply tools/travis/build_byond.sh]: TGM is already checked on the script though.

File: tools/travis/build_byond.sh [hunk 3]
```diff
@@ -5,29 +5,13 @@ set -e
...
 if [ "$BUILD_TOOLS" = false ]; then
-	if grep '"aaa" = \(.+\)' _maps/**/*.dmm;	then
-    	echo "Non-TGM formatted map detected. Please convert it using Map Merger!"
-    	exit 1
-	fi;
-	if grep 'step_[xy]' _maps/**/*.dmm;	then
-    	echo "step_[xy] variables detected in maps, please remove them."
-    	exit 1
-	fi;
 	if grep 'pixel_[xy] = 0' _maps/**/*.dmm;	then
```
Comment (ShizCalev) [inline tools/travis/build_byond.sh]: can lose this one.

File: tools/travis/build_byond.sh (hunk 1 above)
Comment (ShizCalev) [reply tools/travis/build_byond.sh]: Oh wait this hooks into travis, nvm

**Comment by KorPhaeron** [comment]
Stale, reopen when it is done

**Pull request closed without merging**